main.go

```diff
@@ -86,9 +86,19 @@ func main() {
 		}
 	}()
 
+	// define a function that reconnects the email client
+	reconnectFn := func() error {
+		_ = mail.Logout() // attempt logout and don't care about the error
+		mail, err = email.NewClient(emailServer, emailUsername, emailPassword)
```

Comment:
nit: Both `mail` and `err` here are pulled in the closure from the surrounding scope. I would personally prefer to add them as separate arguments, so it's clear what is happening to them. I emphasize that it's a personal preference because I can't point to anything bad that is going to directly happen because of this. I've just had a bunch of bad experiences with bad closures in Node.js and I'm wary of them. :)

Reply:
Well that's the intention right, the mail client is passed by reference and this is essentially swapping it out.
Alternatively, the client could get initiated in the `fetcher` module itself. I didn't do that because that breaks the pattern a bit and it's overall less consistent with how the code is organized. I figured the `reconnectFn` wasn't that bad.